### ISSUE TITLE
After upgrade to 1.36.1, VS Code freezes twice for a few seconds every 2-3 minutes

### ISSUE THREAD

**Issue by bernardoct** [microsoft/vscode#78278]
Issue Type: <b>Performance Issue</b>

After upgrade to 1.36.1, VS Code freezes for a few seconds every 2-3 minutes. The freeze is actually comprised of two few-seconds-long freezes with a fraction of a second between them in which the cursor moves or text that was typed during the first freeze partly shows up.

I have been using VS code to develop Python code, and have tried to disable all my extensions (Python and non-Python related) to no avail. Below is the output of code --status:

Version:          Code 1.36.1 (2213894ea0415ee8c85c5eea0d0ff81ecc191529, 2019-07-08T22:59:35.033Z)
OS Version:       Windows_NT x64 10.0.18362
CPUs:             Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz (4 x 2904)
Memory (System):  15.84GB (6.98GB free)
VM:               0%
Screen Reader:    no
Process Argv:     
GPU Status:       2d_canvas:                     enabled
                  flash_3d:                      enabled
                  flash_stage3d:                 enabled
                  flash_stage3d_baseline:        enabled
                  gpu_compositing:               enabled
                  multiple_raster_threads:       enabled_on
                  native_gpu_memory_buffers:     disabled_software
                  oop_rasterization:             disabled_off
                  protected_video_decode:        enabled
                  rasterization:                 enabled
                  skia_deferred_display_list:    disabled_off
                  skia_renderer:                 disabled_off
                  surface_synchronization:       enabled_on
                  video_decode:                  enabled
                  viz_display_compositor:        disabled_off
                  webgl:                         enabled
                  webgl2:                        enabled

CPU %	Mem MB	   PID	Process
    0	    77	  3848	code main
    1	   183	  3328	   window (paxis_paper3.py - reeval - Visual Studio Code)
    0	     6	  5308	     console-window-host (Windows internal process)
    0	    36	 12392	     searchService
    0	     4	 15480	     C:\WINDOWS\System32\cmd.exe
    0	    34	  8796	       electron_node cli.js 
    0	    52	  4984	         "C:\Program Files\Microsoft VS Code\Code.exe" --status
    0	    14	 16356	     watcherService 
    0	    11	  2888	       console-window-host (Windows internal process)
    0	   180	 16776	     extensionHost
    0	   299	 15140	       C:\Users\Bernardo\.vscode\extensions\ms-python.python-2019.6.24221\languageServer.0.3.43\Microsoft.Python.LanguageServer.exe
    0	    11	 15148	         console-window-host (Windows internal process)
    0	    13	  3560	   electron-crash-reporter
    0	   223	  4640	   gpu-process
    0	    80	  7208	   shared-process

Workspace Stats: 
|  Window (paxis_paper3.py - reeval - Visual Studio Code)
|    Folder (reeval): 5441 files
|      File types: csv(5007) tar_part00(116) tar_part01(116) tar(116)
|                  tar_part02(60) py(9) pyc(8) json(2) tar_part03(2) npy(1)
|      Conf files: launch.json(1) settings.json(1)
|      Launch Configs: python

VS Code version: Code 1.36.1 (2213894ea0415ee8c85c5eea0d0ff81ecc191529, 2019-07-08T22:59:35.033Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz (4 x 2904)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_deferred_display_list: disabled_off<br>skia_renderer: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.84GB (7.25GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    2	    77	 14812	code main
    0	   194	   268	   window (paxis_paper3.py - reeval - Visual Studio Code)
    0	     3	  6136	     C:\WINDOWS\System32\cmd.exe
    0	    71	 11284	     extensionHost
    0	    14	 15048	     watcherService 
    0	    11	  6128	       console-window-host (Windows internal process)
    0	     5	 16708	     console-window-host (Windows internal process)
    0	    13	  1904	   electron-crash-reporter
    2	    67	  7128	   shared-process
    0	   288	 15228	   gpu-process
    0	    69	 15276	   window (Issue Reporter)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (paxis_paper3.py - reeval - Visual Studio Code)
|    Folder (reeval): 5441 files
|      File types: csv(5007) tar_part01(116) tar_part00(116) tar(116)
|                  tar_part02(60) py(9) pyc(8) tar_part03(2) json(2) npy(1)
|      Conf files: launch.json(1) settings.json(1)
|      Launch Configs: python;
```

</details>
Extensions: none

**Comment by joaomoreno** [comment]
Can you reproduce the issue again, but first:

- Open `Developer: Show Running Extensions`
- Click the `Start Extension Host Profile` button at the top right, it's the circle
- Reproduce the issue
- Stop profile

You can show me a screenshot of the profile result. You can also send me the profiling results.

**Comment by bernardoct** [comment]
João, 

Here are the profiling results. The problem was observed within less than 2 minutes and all extensions were disabled.

[CPU-20190802T124950.275Z.cpuprofile.txt](https://github.com/microsoft/vscode/files/3461416/CPU-20190802T124950.275Z.cpuprofile.txt)

**Comment by joaomoreno** [comment]
Nothing stands out in that profile, it indicates no freeze.

Here is a **checklist** to find out more about the freeze/slow/crash
- [ ] it reproduces with extensions disabled (`code --disable-extensions` from the command line - NOT the integrated terminal of Code)
    - if it does not reproduce with extensions disabled, you can stop testing and report this!
- [ ] it was not happening in the previous stable release
- [ ] it reproduces in insiders builds (get it from [here](http://code.visualstudio.com/Download#insiders))
- [ ] there is no suspicious output when running from the command line (`code --verbose`)
- [ ] there is no suspicious output in the window itself (Help | Toggle Developer Tools | Console)
- [ ] it reproduces on Windows, Linux and Mac
- [ ] it reproduces after deleting `<appdata>/User/keybindings.json` and `<appdata>/User/settings.json` (backup first! see below for paths)
- [ ] it reproduces when opening empty (`code -n` from the command line)
- [ ] it reproduces opening any folder
- [ ] it reproduces opening any file
- [ ] it reproduces when only having a single window open
- [ ] it reproduces by just using Code without external tools running on the same folder
- [ ] it reproduces from a workspace that is not under Git version control
- [ ] it reproduces independently if a file is opened or not after startup
- [ ] it reproduces when running with GPU disabled (`code --disable-gpu`)
- [ ] it reproduces when being offline
- [ ] it reproduces when disabling the file watching (set `files.watcherExclude` to `"**/**": true`)
- [ ] it reproduces after uninstalling and reinstalling
- [ ] it reproduces after deleting the application data directory (backup first! see below for paths)
- [ ] I have not played around with permissions or ran Code as administrator
- [ ] I am seeing a single CPU process spin high (please share the full command line arguments)
- [ ] I am seeing Code consuming lots of memory in a short time

**Application Data Directory for VS Code:**
* Windows: `C:\Users\<user>\AppData\Roaming\Code`
* Mac: `/Users/<user>/Library/Application Support/Code`
* Linux: `/home/<user>/.config/Code`

**Application Data Directory for VS Code Insiders:**
* Windows: `C:\Users\<user>\AppData\Roaming\Code - Insiders`
* Mac: `/Users/<user>/Library/Application Support/Code - Insiders`
* Linux: `/home/<user>/.config/Code - Insiders`

**Comment by bernardoct** [comment]
I have tried a number of the of items in the checklist on my Windows 10 laptop (i7-7500U), though VS Code kept freezing every few minutes. I kept my task monitor open and took screenshots of both my screens when it froze as I tried items from the checklist -- the screenshots include the command I used to open VS Code. In all cases, I noticed increased CPU usage when VS Code freezes. I also noticed the same issue happening on my desktop Ubuntu 18.04 (Xeon E5-1620). Attached are a few screenshots of the both Windows and Ubuntu systems. The code I'm running is on a private GitHub repository, which I would be happy to give you access to should it be useful to do so. Both computers are running VS Code v1.36.1. 

Is there anything in particular I should try next?

![VSCode_freeze_screenshots](https://user-images.githubusercontent.com/9001389/62746929-0b21e000-ba20-11e9-94ba-25ee115fda19.png)
![Screenshot from 2019-08-08 17-49-27](https://user-images.githubusercontent.com/9001389/62747004-63f17880-ba20-11e9-98c8-7d61b8d09578.png)

**Comment by joaomoreno** [comment]
@bpasero Bring in those :eyes: of yours

**Comment by bpasero** [comment]
We are building exploration builds that use a much newer version of our UI framework (Electron version 6.0.x). I wonder if this issue reproduces with one of these builds, could you try? Download:
* macOS: [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/VSCode-darwin-exploration.zip)
* Linux (DEB): [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/code-exploration_1.37.0-1564556412_amd64.deb)
* Linux (RPM): [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/code-exploration-1.37.0-1564556601.el7.x86_64.rpm)
* Windows: [Download](https://az764295.vo.msecnd.net/exploration/960208fdf9aa06e6892aa14c9c84ddd607fbeb2f/VSCodeUserSetup-x64-1.37.0-exploration.exe)

**Comment by bernardoct** [comment]
I just tried suggested exploration build on my Ubuntu 18.04 desktop. I got a slightly different freeze this time, in which the text cursor keeps blinking and the system monitor shows no increase in CPU usage, though I still can't type and VS Code gets partly unresponsive to the mouse, which kept the text symbol as I moved the cursor all throughout VS Code screen. 

Anything else I should try?

**Comment by bernardoct** [comment]
No freeze on Windows 10 after installing the exploration build.

**Comment by bernardoct** [comment]
Update: I have now been using VS Code v1.37.1 on Windows for Latex editing and the freeze still happens.

**Comment by bernardoct** [comment]
Update: I have used VS Code v1.38 today for a whole day in my Ubuntu desktop and the freezes are not observed anymore. This probably settles the issue.

**Comment by bernardoct** [comment]
I installed the released v1.38.0, instead of the exploration build, on my Windows 10 laptop and the issue started back.

**Comment by bpasero** [comment]
Can you follow this wiki and post the results back here to understand more about why performance is so bad for you: https://github.com/Microsoft/vscode/wiki/Performance-Issues

Thanks!
